src/ctia/stores/es/crud.clj

```diff
@@ -117,32 +112,48 @@ It returns the documents with full hits meta data including the real index in wh
    es-params]
   (first (get-docs-with-indices conn-state [_id] es-params)))
 
-
 (defn ^:private prepare-opts
   [{:keys [props]}
    {:keys [refresh]}]
   {:refresh (or refresh
                 (:refresh props)
                 "false")})
 
+(s/defn bulk-schema
+  [Model :- (s/pred map?)]
+  (st/optional-keys
+   {:create [Model]
+    :index [Model]
+    :update [(st/optional-keys Model)]
+    :delete [s/Str]}))
+
+(s/defn ^:private prepare-bulk-doc
+  [{:keys [props]} :- ESConnState
+   mapping :- s/Keyword
+   doc :- (s/pred map?)]
+  (assoc doc
+         :_id (:id doc)
+         :_index (:write-index props)
+         :_type (name mapping)))
+
 (defn handle-create
   "Generate an ES create handler using some mapping and schema"
   [mapping Model]
   (let [coerce! (coerce-to-fn (s/maybe Model))]
     (s/fn :- (s/maybe [Model])
-      [{:keys [conn props] :as conn-state} :- ESConnState
+      [{:keys [conn] :as conn-state} :- ESConnState
        docs :- [Model]
        _ident
        es-params]
-      (let [prepare-doc #(assoc %
-                                :_id (:id %)
-                                :_index (:write-index props)
-                                :_type (name mapping))]
+      (let [prepare-doc (partial prepare-bulk-doc conn-state mapping)
+            prepared (doall
+                      (map prepare-doc docs))]
         (try
-          (map #(build-create-result % coerce!)
-               (ductile.doc/bulk-create-doc conn
-                                            (map prepare-doc docs)
-                                            (prepare-opts conn-state es-params)))
+          (ductile.doc/bulk-index-docs conn
+                                       prepared
+                                       (prepare-opts conn-state es-params)
+                                       50000)
```

Comment:
Why this value? Shouldn't you put this magic number in the configuration? Or perhaps get this value depending on some ES configuration value?

Reply:
another trace of a dev fight. I remove it, it is solved with https://github.com/threatgrid/ductile/pull/16